src/mbgl/gl/location_component_layer_impl.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <array>
+#include <mbgl/gl/location_component_layer.hpp>
+#include <mbgl/gl/location_component_layer_properties.hpp>
+#include <mbgl/map/transform_state.hpp>
+#include <mbgl/renderer/paint_parameters.hpp>
+#include <mbgl/style/layer_impl.hpp>
+#include <mbgl/style/layer_properties.hpp>
+#include <mbgl/util/color.hpp>
+#include <mbgl/util/geo.hpp>
+#include <memory>
+#include <string>
+
+namespace mbgl {
+
+class TransformState;
+
+namespace style {
+
+struct LocationComponentRenderParameters {
+    LocationComponentRenderParameters(){};
+    LocationComponentRenderParameters(const TransformParameters &tp) : state(&tp.state) {}
```

Comment:
this constructor will leave a bunch of fields unassigned, is it expected?

Reply:
Yes, more or less. The data will be filled in RenderLocationComponentLayer::render before sending it through.